### ISSUE TITLE
Need fix, that display only build under Linux works.

### ISSUE THREAD

**Issue by FslNopper** [KhronosGroup/Vulkan-LoaderAndValidationLayers#926]
We need a build configuration, where a Linux build without XCB and XLIB but with VK_KHR_display extension enabled is possible. Added pull request for SDK 1.0.21 (which can be dropped) and master branch:

Fix for 1.0.21:
https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/pull/893

Fix for master branch:
https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/pull/925

**Comment by karl-lunarg** [comment]
I'd like to pick up this fix in the next day or so, but I would also like to add some tests to make sure that this doesn't break if someone modifies the cmake files again, since we don't test this config regularly.

Can you suggest a test that can be run on a Linux/X machine to test this build config?   About all I can think of is to build with the flags to disable XLIB and XCB support to make sure it still builds, and then run vulkaninfo and make sure we don't see the KHR Xlib abd Xcb surface extensions.

**Comment by McNopper** [comment]
Thanks for having a look at this.

You can build e.g. under Ubuntu 16.04 - without a cross compiler enabled - the following way:

cmake -H. -Bdbuild -DCMAKE_BUILD_TYPE=Debug -DBUILD_WSI_XCB_SUPPORT=OF
F -DBUILD_WSI_XLIB_SUPPORT=OFF -DCMAKE_SYSTEM_PROCESSOR=arm

It builds and also vulkaninfo can be executed. The other examples are currently not build, as the VK_KHR_display code is missing in the examples. Either VeriSilicon or we will implement this at a later point of time.

We use this build configuration to cross compile for our i.MX8 target having Linux with no X11 on it - but the WSI extension VK_KHR_display is available and working.

From my tests, the changes do not harm the current available build configurations.

PS: Writing form my private account - FslNopper is me as well :-)

PS PS: You can assume, that VeriSiliicon and NXP will test this configuration for the future.

**Comment by karl-lunarg** [comment]
Another complication is that many of these files propagate to the https://github.com/LunarG/VulkanTools and https://github.com/LunarG/VulkanSamples repos.  In places where these changes overlap, like disabling the build of demos like cube, things are pretty straightforward.  But in other places, there will be additional items in VT and VS that need similar disabling.  For example, would it be correct to disable the build of all executables in VulkanSamples?

In short, I'll need to consider the impact of these changes into these other two repos as well.  I don't know how much you care about these other two repos, but I'll still have to handle things somehow.

**Comment by FslNopper** [comment]
For now, I would suggest to disable the build of the VulkanSamples in the other repos. In a next step, the necessary code to use the VK_KHR_display extension can be added.
Most important is, that the loader and validation layers do build. Regarding the samples, let's see if VeriSilicon and/or we can commit our changes as soon as possible.

**Comment by McNopper** [comment]
In my opinion, this is not just an enhancement.
It is a bug, as building for just having the WSI extension "VK_KHR_display" present is valid and not an optional feature.

Hope this fix makes it into the next LuanrG SDK as well.

**Comment by karl-lunarg** [comment]
https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/pull/925 closed.
